test/second_q/mappers/test_bosonic_linear_mapper.py

```diff
@@ -0,0 +1,188 @@
+
+""" Test Bosonic Linear Mapper """
+
+import unittest
+
+from test import QiskitNatureTestCase
+
+from ddt import ddt, data, unpack
+import numpy as np
+
+from qiskit.quantum_info import SparsePauliOp
+from qiskit_nature.second_q.operators import BosonicOp
+from qiskit_nature.second_q.mappers import BosonicLinearMapper
+from qiskit_nature import settings
+
+
+@ddt
+class TestBosonicLinearMapper(QiskitNatureTestCase):
+    """Test Bosonic Linear Mapper"""
+
+    # Define some useful coefficients
+    sq_2 = np.sqrt(2)
+
+    bos_op1 = BosonicOp({"+_0": 1})
+    # Using: truncation = 1
+    ref_qubit_op1_tr1 = SparsePauliOp(["XX", "YY", "YX", "XY"], coeffs=[0.25, 0.25, -0.25j, 0.25j])
+    # Using: truncation = 2
+    ref_qubit_op1_tr2 = SparsePauliOp(
+        ["IXX", "IYY", "IYX", "IXY", "XXI", "YYI", "YXI", "XYI"],
+        coeffs=[0.25, 0.25, -0.25j, 0.25j, sq_2 / 4, sq_2 / 4, -1j * sq_2 / 4, 1j * sq_2 / 4],
+    )
+
+    bos_op2 = BosonicOp({"-_0": 1})
+    # Using: truncation = 1
+    ref_qubit_op2_tr1 = SparsePauliOp(["XX", "YY", "YX", "XY"], coeffs=[0.25, 0.25, 0.25j, -0.25j])
+    # Using: truncation = 2
+    ref_qubit_op2_tr2 = SparsePauliOp(
+        ["IXX", "IYY", "IYX", "IXY", "XXI", "YYI", "YXI", "XYI"],
+        coeffs=[0.25, 0.25, 0.25j, -0.25j, sq_2 / 4, sq_2 / 4, 1j * sq_2 / 4, -1j * sq_2 / 4],
+    )
+
+    bos_op3 = BosonicOp({"+_1": 1})
+    # Using: truncation = 1
+    ref_qubit_op3_tr1 = SparsePauliOp(
+        ["XXII", "YYII", "YXII", "XYII"], coeffs=[0.25, 0.25, -0.25j, 0.25j]
+    )
+    # Using: truncation = 2
+    ref_qubit_op3_tr2 = SparsePauliOp(
+        ["IXXIII", "IYYIII", "IYXIII", "IXYIII", "XXIIII", "YYIIII", "YXIIII", "XYIIII"],
+        coeffs=[0.25, 0.25, -0.25j, 0.25j, sq_2 / 4, sq_2 / 4, -1j * sq_2 / 4, 1j * sq_2 / 4],
+    )
+
+    bos_op4 = BosonicOp({"-_1": 1})
+    # Using: truncation = 1
+    ref_qubit_op4_tr1 = SparsePauliOp(
+        ["XXII", "YYII", "YXII", "XYII"], coeffs=[0.25, 0.25, 0.25j, -0.25j]
+    )
+    # Using: truncation = 2
+    ref_qubit_op4_tr2 = SparsePauliOp(
+        ["IXXIII", "IYYIII", "IYXIII", "IXYIII", "XXIIII", "YYIIII", "YXIIII", "XYIIII"],
+        coeffs=[0.25, 0.25, 0.25j, -0.25j, sq_2 / 4, sq_2 / 4, 1j * sq_2 / 4, -1j * sq_2 / 4],
+    )
+
+    bos_op5 = BosonicOp({"+_0 -_0": 1})
+    # Using: truncation = 1
+    ref_qubit_op5_tr1 = SparsePauliOp(["II", "ZZ", "ZI", "IZ"], coeffs=[0.25, -0.25, -0.25, 0.25])
+    # Using: truncation = 2
+    ref_qubit_op5_tr2 = SparsePauliOp(
+        ["III", "IZZ", "IZI", "IIZ", "ZZI", "ZII"], coeffs=[0.75, -0.25, 0.25, 0.25, -0.5, -0.5]
+    )
+
+    bos_op6 = BosonicOp({"-_0 +_0": 1})
+    # Using: truncation = 1
+    ref_qubit_op6_tr1 = SparsePauliOp(["II", "ZZ", "ZI", "IZ"], coeffs=[0.25, -0.25, +0.25, -0.25])
+    # Using: truncation = 2
+    ref_qubit_op6_tr2 = SparsePauliOp(
+        ["III", "IZZ", "IZI", "IIZ", "ZZI", "ZII"], coeffs=[0.75, -0.25, -0.25, -0.25, -0.5, 0.5]
+    )
+
+    # fmt: off
```

Comment:
Can you please limit the format disabling to really just the few lines where its needed?

Reply:
Done